adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralBannerRenderer.m

```diff
@@ -0,0 +1,131 @@
+#import "GADMediationAdapterMintegralConstants.h"
+#include <stdatomic.h>
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKBanner/MTGBannerAdView.h>
+#import <MTGSDKBanner/MTGBannerAdViewDelegate.h>
+
+@interface GADMAdapterMintegralBannerRenderer ()
+<GADMediationBannerAd,
+MTGBannerAdViewDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralBannerRenderer{
+    /// Data used to render an banner ad.
+    GADMediationBannerAdConfiguration *_adConfiguration;
+    
+    // The completion handler to call when the ad loading succeeds or fails.
+    GADMediationBannerLoadCompletionHandler _adLoadCompletionHandler;
+    
+    // An ad event delegate to invoke when ad rendering events occur.
+    __weak id<GADMediationBannerAdEventDelegate> _adEventDelegate;
+    
+    // The Mintegral banner ad.
+    MTGBannerAdView *_bannerAdView;
+    
+}
+
+- (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                   completionHandler:
+(nonnull GADMediationBannerLoadCompletionHandler)completionHandler{
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationBannerLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationBannerAdEventDelegate>(
+                                                                      _Nullable id<GADMediationBannerAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationBannerAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+        
+    NSString *unitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    UIViewController *rootViewController = adConfiguration.topViewController;
+
+    if ([GADMAdapterMintegralUtils isEmpty:unitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if ([GADMAdapterMintegralUtils isEmpty:adConfiguration.bidResponse]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"bid token cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    _bannerAdView = [[MTGBannerAdView alloc]initBannerAdViewWithAdSize:adConfiguration.adSize.size placementId:placementId unitId:unitId rootViewController:rootViewController];
+    _bannerAdView.delegate = self;
+    _bannerAdView.autoRefreshTime = 0;
```

Comment:
Is this going to override the banner refresh time set by the publisher in the AdMob UI?

Reply:
no,this is just to close mintegral auto refresh ads